shared/hashutil/deposit_hash.go

```diff
@@ -30,13 +30,13 @@ func DepositHash(dep *pb.DepositData) ([32]byte, error) {
 
 	var zeroBytes [32]byte
 
-	pubkeyRoot := HashSha256(append(dep.Pubkey, zeroBytes[:15]...))
-	sigHash := HashSha256(dep.Signature[:63])
-	sigZeroBytesHash := HashSha256(append(dep.Signature[63:95], zeroBytes[:]...))
+	pubkeyRoot := HashSha256(append(dep.Pubkey, zeroBytes[:16]...))
+	sigHash := HashSha256(dep.Signature[:64])
+	sigZeroBytesHash := HashSha256(append(dep.Signature[64:96], zeroBytes[:]...))
```

Comment:
Looks great! Just curious, why was this change needed? Aren't these 0 index based?

Reply:
Yeah they are, but this didnt actually conform to the spec. So it took the first 15 bytes instead of 16, etc. It was the main reason the trie didnt work, the wrong byte buffers were being hashed, leading to different deposit roots